src/presentation/http/router/noteSettings.test.ts

```diff
@@ -281,4 +281,61 @@ describe('NoteSettings API', () => {
 
     test.todo('Return 403 when user authorized, but not member of the team');
   });
+
+  describe('PATCH /note-teams/new-role/:newRole', () => {
+    test('Update team member role by user id and note id', async () => {
+      await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${global.auth(1)}`,
+        },
+        url: '/note-settings/new-role/Pq1T9vc23Q/1/write',
+      });
+
+      const team = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${global.auth(1)}`,
+        },
+        url: '/note-settings/Pq1T9vc23Q/team',
+      });
+
+      if (team?.json() !== undefined) {
+        expect(team?.json()).toStrictEqual([
+          {
+            'id': 2,
+            'noteId': 2,
+            'userId': 1,
+            'role': 1,
+          },
+        ]);
+      }
+    });
+
+    test('Returns status code 200 and new role if it was patched even if user already has a passing role', async () => {
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${global.auth(1)}`,
+        },
+        url: '/note-settings/new-role/Pq1T9vc23Q/1/write',
+      });
+
+      expect(response?.statusCode).toBe(200);
+      expect(response?.body).toBe('write');
+    });
+
+    test('Returns status code 404 and "User in team is not found" message if no such a note exists', async () => {
```

Comment:
test case title does not match actual behavior

Reply:
fixed